app/controllers/api/motors_controller.rb

```diff
@@ -22,15 +22,28 @@ def new
 
   # POST /motors
   def create
-    @location = Location.find_or_create_by(name: params[:location_name])
+    # Get the city_name from the motor parameters
+    city_name = params[:city_name]
 
-    @motor = @location.motors.new(motor_params)
-    @motor.user = current_user
+    # Find an existing StoreLocation by city_name
+    store_location = StoreLocation.find_by(city_name:)
+
+    if store_location.nil?
+      # If it doesn't exist, create a new StoreLocation
+      store_location = StoreLocation.create(city_name:)
```

Comment:
Nice job updating the motors_controller :ok_hand: , But there is an issue you need to fix on line 29 and 33. it is giving this error when opening the motors API on the browser
![Screenshot from 2023-10-07 21-06-13](https://github.com/binodbhusal/final_capstone_backend/assets/103933572/3782d1a9-b980-4ea9-85ec-ac2c20dac4da)

Kindly, Udate the line 29 to this ``` store_location = StoreLocation.find_by(city_name: city_name) ``` and the line 33 to this ``` store_location = StoreLocation.create(city_name: city_name) ```. That will fix the error :wink:

Reply:
@dieum512 I have no error on my side I checked it. Did you seed the data? I will check again if any.